### ISSUE TITLE
Web dash lags caused by invisible components?

### ISSUE THREAD

**Issue by Romainrob** [SHWotever/SimHub#760]
Bonsoir,

I'm investigating a performance with my iRacing dashboard.
It tends to lag (frequent short hangs) when running on some devices. My Samsung Tab A 2019 tablet in this case.
I seemed smooth on my iPad, and on my PC screen (in html)

I'm aware of wifi and device limitations (and rebooted everything) but I think there is an issue related to inherited visibility not stopping the data of invisible components being pushed. Possibly making things harder for weaker devices.

As a test, I enabled the data update control mode, An ran the same 5 min replay in loop.
My dash main screen has 10 pages, which are layers with exclusive visibility conditions. I left it on page one.
With the dash as it is, all pages present but only page one visible, I got a lot of web dash throttling.
Then on a duplicate dash I deleted the 9 invisible pages, and I was smooth. No throttling.

The hidden pages include maps and charts, which I suspect to be the main culprit. I'll try to narrow it down if possible.

Let me know if can run specific tests to help figure this one out.
Bonne soirée!

**Comment by Romainrob** [comment]
I did more testing, deleting various pages.
The charts seems to be the main, if not only, offenders.
Deleting the input page, containing 5 charts, solves the lag issue.

**Comment by Romainrob** [comment]
I'm done testing for today. I couldn't find a workaround :(.
I don't mind the charts lagging on slower devices, but having them slowing down the whole main screen even when not displayed is a bummer.
If it helps, here are the 2 versions of my dash, with and with out the charts:
https://www.dropbox.com/s/0hx2r4rukwoks38/SimHub%20Charts%20Test%20-%20Ticket%20%23760.7z?dl=1

**Comment by SHWotever** [comment]
Hello !

I checked and the rendering is suspended when on another page, however I suspect that maintaining the render state of the component in the browser is maybe a bit too much, so I unloaded the whole rendering part and reloaded it when it's back to visibility. It's a bit of overhead when changing the page, but I hope it will kill lags lags.

**Comment by Romainrob** [comment]
Thank you!
Count on me to let you know if it helped ;)
Will this apply to the charts only or any components?
I noticed some discrepancy between the web and desktop mode, when going back to a screen. For example some changed, isincreasing.... based formulas would be erroneously true after a screen change (in html mode only I believe), as if it restarted from the state the data was in last time the screen was displayed.
After the initial headache I was able to work around this by moving some formulas outside the dash, but I'm curious if this change will make the web and desktop mode more uniform on this front.
